bee-node/bee-node/src/inx.rs

```diff
@@ -0,0 +1,606 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use bee_ledger::{
+    types::{ConsumedOutput, CreatedOutput, LedgerIndex, Unspent},
+    workers::event::{LedgerUpdated, ReceiptCreated},
+};
+use bee_message::{
+    constant::PROTOCOL_VERSION, milestone::MilestoneIndex, output::OutputId, payload::Payload,
+    semantic::ConflictReason, Message, MessageId,
+};
+use bee_protocol::workers::{
+    event::{MessageProcessed, MessageSolidified},
+    MessageSubmitterError, MessageSubmitterWorker, MessageSubmitterWorkerEvent,
+};
+use bee_runtime::{event::Bus, node::Node, resource::ResourceHandle, worker::Worker};
+use bee_storage::{
+    access::{Exist, Fetch},
+    system::StorageHealth,
+};
+use bee_tangle::{
+    event::{ConfirmedMilestoneChanged, LatestMilestoneChanged},
+    metadata::MessageMetadata,
+    Tangle,
+};
+use inx::{
+    proto,
+    server::{Inx, InxServer},
+    Request, Response, Status,
+};
+use packable::PackableExt;
+use tokio_stream::wrappers::UnboundedReceiverStream;
+
+use crate::{FullNode, NodeStorageBackend};
+
+fn convert_message_id(message_id: &MessageId) -> proto::MessageId {
+    proto::MessageId {
+        id: message_id.as_ref().to_vec(),
+    }
+}
+
+pub struct InxWorker {}
+
+#[async_trait::async_trait]
+impl<B: NodeStorageBackend> Worker<FullNode<B>> for InxWorker {
+    type Config = ();
+
+    type Error = std::convert::Infallible;
```

Comment:
If the server fails for some reason, should we return an error instead of panicking? I feel like yes, so we can prevent corrupting the database. Wdyt?